psycopg/psycopg/types/hstore.py

```diff
@@ -74,6 +83,39 @@ def add_token(s: str) -> None:
         return dumper.dump(data)
 
 
+class BaseHstoreBinaryDumper(RecursiveDumper):
+    format = Format.BINARY
+    encoding: str
```

Comment:
As explained in `_register_hstore()` comment, similarly to the binary loader, the encoding should be an instance property:

```suggestion
class BaseHstoreBinaryDumper(Dumper):
    format = Format.BINARY

    def __init__(self, cls: type, context: AdaptContext | None = None):
        super().__init__(cls, context)
        enc = conn_encoding(self.connection)
        self.encoding = enc if enc != "ascii" else "utf-8"
```